back/server/modules/battleshipRound.js

```diff
@@ -36,25 +71,119 @@ class Map {
             this.Map[coordinate.x][coordinate.y] = { shipId: shipId, isHit: false };
         });
     }
+
+    /**
+     * 
+     * @param {Coordinate[]} shipDefinition 
+     * @returns {boolean} returns if the ship placement would be valid.
+     */
+    isValid(shipDefinition){
+        shipDefinition.forEach(coordinate => { //coordinate should be an array, so runs through each
+            if (this.Map[coordinate.x][coordinate.y].shipId == null) {
+              return false;
+            }
+        });
+        return true;
+    }
+    /**
+     * 
+     * @param {number} shipId should be between 1 and 5
+     */
+    addAIShip(shipId) {
+        const horizontal = (Math.random() >= 0.5)
+        while (1) {
+          
+            let x = Math.trunc(Math.random() * (this.dimensions[0]-1));
+            let y = Math.trunc(Math.random() * (this.dimensions[1]-1));
+            
+            /** @type {Coordinate[]} */
+            let coordinates = [];
+            
+            for (let i = 0; i < shipId; i++) {
+                if (horizontal) {
+                    let _x;
+                    if (x + i >= 10) {
+                        _x = x - (x + i - 10) - 1
+                    } else {
+                        _x = x + i
+                    }
+                    coordinates.push({x:_x, y:y})
+                } else {
+                    let _y;
+					if (y + i >= 10) {
```

Comment:
not sure if it is just my computer but these few lines are really far indented to the right.